### PR TITLE
Support two-tier Collection

### PR DESCRIPTION
**Pull request by normal-coder**
### What does this Pull Request do

* adjust the one-level structure to support the two-level structure
* align filename and Collection Ident

### CODE REVIEW

File: collections.yml [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    - big-data # 大数据
+- ident: development-framework
+  name: Development Framework
+  name_cn: 开发框架
+  items:
+    - android-development-framework # 安卓开发框架
+    - authorization-framework # 权限认证框架
+    - crawler # 网络爬虫框架
+    - css-framework # CSS 框架
+    - deep-learning-frameworks # 深度学习框架
```
Comment (eyehwan) [inline collections.yml:14]: Type of noun, plural not recommended; 分类名词，不建议用复数

File: collections.yml [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    - edge-computing-frameworks # 边缘计算框架
```
Comment (eyehwan) [inline collections.yml:15]: the same as above.

File: collections.yml [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    - ios-framework # iOS 开发框架
+    - node.js-web-framework # Node.js Web 框架
+    - privacy-computing-frameworks # 隐私计算框架
+    - serverless-framework # Serverless 框架
```
Comment (eyehwan) [inline collections.yml:19]: here

File: collections.yml [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    - unit-testing-framework # 单元测试框架
+    - web-frameworks # Web 开发框架
+- ident: devops
+  name: DevOps
+  name_cn: DevOps
+  items:
+    - continuous-deployment-tools # 持续部署工具
+    - continuous-integration-tools # 持续集成工具
+- ident: emerging-technologies
+  name: Emerging Technologies
```
Comment (eyehwan) [inline collections.yml:29]: The same comment as above

File: collections.yml [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    - e2e-testing # 端到端测试
+    - eBPF # eBPF
+    - emacs # Emacs
+    - emoji # Emoji 应用
+    - file-storage # 文件存储
+    - finite-state-machine # 有限状态机（FSM）
+    - flow-chart # 流程图
+    - fonts # 字体
+    - full-text-search-engine # 全文搜索引擎
+    - game-development-kit # 游戏开发套件
```
Comment (eyehwan) [inline collections.yml:184]: Should add Game first level?

File: collections.yml [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    - irc-server # IRC 服务器
+    - kanban # 看板工具
+    - knowledge-graph # 知识图谱
+    - logging # 日志记录管理
+    - mail-client # 邮件客户端
+    - mail-server # 邮件服务器
+    - media-players # 媒体播放器
+    - mind-map # 思维导图
+    - mlops # 自动化机器学习
+    - Monitoring # 系统监控
```
Comment (eyehwan) [inline collections.yml:205]: development-toolkit?

File: collections.yml [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+    - motion-graphics # 动态图形软件
+    - nginx-related # Nginx 相关
+    - nlp # 自然语言处理
```
Comment (eyehwan) [inline collections.yml:208]: Should be added to Machine Learning first level?

File: collections.yml [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+  name_cn: 安全相关
+  items:
+    - application-definition-and-image-build # 应用程序定义和镜像构建
+    - cve-scanning # CVE 漏洞扫描
+    - dependency-analysis # 依赖分析
+    - dependency-management # 依赖管理
+    - security-and-compliance # 安全与合规工具
+    - security-audit # 安全审计
+- ident: others
+  name: Others
```
Comment (eyehwan) [inline collections.yml:151]: Too many others...

File: collections.yml (hunk 1 above)
Comment (normal-coder) [reply collections.yml]: Agree.

File: collections.yml [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+    - video-editor # 视频剪辑
+    - virtual-machine # 虚拟机
+    - virtual-reality # 虚拟现实
+    - vlc-player # VLC 播放器
+    - voip-and-video-conferencing # VoIP 和视频会议
+    - web-servers # Web 服务器
+    - webassembly # WebAssembly(WASM)
+    - webrtc-server # WebRTC 服务器
+    - workflow-scheduler # 工作流调度器
+    - wysiwyg-editor # 即见即所得编辑器
```
Comment (hncslwx) [inline collections.yml:253]: 【这是一条测试】建议放到“”办公软件类 - 文字编辑软件“”

**Review by eyehwan — APPROVED** [review]

**Pull request merged**